src/Stratis.Bitcoin.Features.RPC/Controllers/FullNodeController.cs

```diff
@@ -91,30 +94,69 @@ public Task Stop()
+        /// When called without a blockhash argument, getrawtransaction will return the transaction if it is in the mempool, or if -txindex is enabled and the transaction is in a block in the blockchain.</remarks>
         [ActionName("getrawtransaction")]
         [ActionDescription("Gets a raw, possibly pooled, transaction from the full node.")]
-        public async Task<TransactionModel> GetRawTransactionAsync(string txid, int verbose = 0)
+        public async Task<TransactionModel> GetRawTransactionAsync(string txid, int verbose = 0, string blockHash = null)
         {
-            uint256 trxid;
-            if (!uint256.TryParse(txid, out trxid))
-                throw new ArgumentException(nameof(txid));
+            Guard.NotEmpty(txid, nameof(txid));
 
-            Transaction trx = this.pooledTransaction != null ? await this.pooledTransaction.GetTransaction(trxid) : null;
+            if (!uint256.TryParse(txid, out uint256 trxid))
+            {
+                throw new ArgumentException(nameof(trxid));
+            }
 
-            if (trx == null)
+            uint256 hash = null;
+            if (!string.IsNullOrEmpty(blockHash) && !uint256.TryParse(blockHash, out hash))
             {
-                trx = this.blockStore != null ? await this.blockStore.GetTransactionByIdAsync(trxid) : null;
+                throw new ArgumentException(nameof(blockHash));
             }
 
-            if (trx == null)
-                return null;
+            // Special exception for the genesis block coinbase transaction.
+            if (trxid == this.Network.GetGenesis().GetMerkleRoot().Hash)
+            {
+                throw new RPCServerException(RPCErrorCode.RPC_INVALID_ADDRESS_OR_KEY, "The genesis block coinbase is not considered an ordinary transaction and cannot be retrieved.");
+            }
+
+            Transaction trx = null;
+            ChainedHeaderBlock chainedHeaderBlock = null;
 
+            if (hash == null)
+            {
+                // Look for the transaction in the mempool, and if not found, look in the indexed transactions.
+                trx = (this.pooledTransaction == null ? null : await this.pooledTransaction.GetTransaction(trxid)) ??
+                      await this.blockStore.GetTransactionByIdAsync(trxid).ConfigureAwait(false);
+
+                if (trx == null)
+                {
+                    throw new RPCServerException(RPCErrorCode.RPC_INVALID_ADDRESS_OR_KEY, "No such mempool transaction. Use -txindex to enable blockchain transaction queries.");
+                }
+            }
+            else
+            {
+                // Retrieve the block specified by the block hash.
+                await this.ConsensusManager.GetOrDownloadBlocksAsync(new List<uint256> { hash }, block => { chainedHeaderBlock = block; });
```

Comment:
Interesting, I am not convinced download is correct here, you better use `GetBlockDataAsync` 

If part of pruned node perhaps then it makes sense however then the delegate needs to be waited if we do queue the block for download the await on the method itself is not enough, but for that I would recommend a new method all together.

Reply:
Changed.